components/ui/service-icon.tsx

```
@@ -0,0 +1,168 @@
+
+    // State for dark mode - initialized false but updated in useEffect
+    const [isDarkMode, setIsDarkMode] = useState(false);
+
+    // Effect to detect and update color scheme
+    useEffect(() => {
+        // Check if window exists (client-side only)
+        if (typeof window !== 'undefined') {
+            // Set initial value
+            setIsDarkMode(
+                window.matchMedia &&
+                window.matchMedia('(prefers-color-scheme: dark)').matches
```

Biome (1.9.4)

[error] Change to an optional chain.

Unsafe fix: Change to an optional chain.


(lint/complexity/useOptionalChain)